### ISSUE TITLE
buildah push: error locating image

### ISSUE THREAD

**Issue by glasvandrare** [projectatomic/buildah#324]
Hello!
I've discovered something a bit strange with the push command, for some reason it can't find the image when I use the ID Sha, but it works when I use the IMAGE NAME reference:

```
buildah push $ubuntu_image oci:/tmp/image:test
error pushing image "0a85c244f0fb" to "oci:/tmp/image:test": error locating image "0a85c244f0fb": image not known
```

```
buildah push docker.io/library/ubuntu:latest oci:/tmp/image:test
Getting image source signatures
Copying blob sha256:ae79f251470513c2a0ec750117a81f2d58a50727901ca416efecf297b8a03913
 45.41 MiB / 45.41 MiB [====================================================] 0s
Copying blob sha256:c59d01a7e4caf1aba785eb33192fec3f96e4ab01975962bcec10f4989a6edcc6
 848 B / 848 B [============================================================] 0s
Copying blob sha256:41ba73a9054d231e1f555c40a74762276900cc6487f5c6cf13b89c7606635c67
 619 B / 619 B [============================================================] 0s
Copying blob sha256:f1bbfd495cc1112b503350686641ee4fa2cea8ccd13fb8a8a302c81dae61d418
 854 B / 854 B [============================================================] 0s
Copying blob sha256:0c346f7223e24b517358f52c4a3f5f9af1c86e5ddeaee5659c8889846e46d1e2
 169 B / 169 B [============================================================] 0s
Copying config sha256:c2a063934a23a229b454b6ed5a62f444b2a0762e7db4fcb3a9f1c9c9c6a53f0e
 2.64 KiB / 2.64 KiB [======================================================] 0s
Writing manifest to image destination
Storing signatures

```
```
buildah images
IMAGE ID             IMAGE NAME                                               CREATED AT             SIZE
0a85c244f0fb         docker.io/library/ubuntu:latest                          Nov 4, 2017 09:45      116.4 MB

```

What am I missing? Isn't IMAGE ID supported as a reference, or is this a bug? 

I'm running Ubuntu 16.04, 
buildah version
Version:       0.5
Go Version:    go1.9
Image Spec:    1.0.0
Runtime Spec:  1.0.0
Git Commit:    d031232
Built:         Mon Nov 13 08:12:58 2017
OS/Arch:       linux/amd64

**Comment by TomSweeneyRedHat** [comment]
@glasvandrare thanks for the problem report!  I'm assuming that in your example the $ubuntu_image variable was set to your image's id '0a85c244f0fb'.  I tried to replicate the issue using an image id like '0a85c244f0fb' and had the same results as you did.  

I believe this is a bug, but will have to check with @nalind.  If it's not a bug in the code, then the man page needs to be touched up and clarified.

FWIW, when I use 'buildah push' I generally use the full name of the image as you did in your successful push.

**Comment by glasvandrare** [comment]
Yes, exactly, and I didn't know if it was a documentation issue, or a functional issue (although I don't see why a reference couldn't be either IMAGE ID or IMAGE NAME).

**Comment by grafarg** [comment]
I ran into the same issue based on instructions from the `man buildah-push`. Note the SYNOPSIS reference to `imageID` (not `imageName`).

`buildah-push"1"(June 2017)                                                             buildah-push"1"(June 2017)

NAME
       buildah push - Push an image from local storage to elsewhere.

SYNOPSIS
       buildah push [options [...]] imageID [destination]
`

**Comment by TomSweeneyRedHat** [comment]
I think a similar fix to what @baude did recently in kpod at: https://github.com/projectatomic/libpod/pull/76 would do the trick.

**Comment by rhatdan** [comment]
@TomSweeneyRedHat Could you open a PR.

**Comment by TomSweeneyRedHat** [comment]
Will do.

**Comment by TomSweeneyRedHat** [comment]
Fixed via #341.  @glasvandrare if you find that's not the case in your testing, please let us know.  Thanks for the report, much appreciated!
